src/assets/scripts/client/aircraft/AircraftInstanceModel.js

```diff
@@ -2272,49 +1259,146 @@ export default class Aircraft {
+
+        if (isInsideAirspace !== this.inside_ctr) {
+            this.crossBoundary(isInsideAirspace);
+        }
+    }
+
+    /**
+     * This turns the aircraft if it is not on the ground and has not arived at its destenation
+     * @for AircraftInstanceModel
+     * @method updateAircraftTurnPhysics
+     */
+    updateAircraftTurnPhysics() {
+        // Exits eary if the airplane is on the ground or at its destenation
+        if (this.isOnGround() && this.heading === this.target.heading) {
+            return;
+        }
         // TURNING
         // this.target.heading = radians_normalize(this.target.heading);
-        if (!this.isOnGround() && this.heading !== this.target.heading) {
-            // Perform standard turns 3 deg/s or 25 deg bank, whichever
-            // requires less bank angle.
-            // Formula based on http://aviation.stackexchange.com/a/8013
-            const turn_rate = clamp(0, 1 / (this.speed / 8.883031), 0.0523598776);
-            const turn_amount = turn_rate * window.gameController.game_delta();
-            const offset = angle_offset(this.target.heading, this.heading);
-
-            if (abs(offset) < turn_amount) {
-                this.heading = this.target.heading;
-            } else if ((offset < 0 && this.target.turn === null) || this.target.turn === 'left') {
-                this.heading -= turn_amount;
-            } else if ((offset > 0 && this.target.turn === null) || this.target.turn === 'right') {
-                this.heading += turn_amount;
-            }
+        // Perform standard turns 3 deg/s or 25 deg bank, whichever
+        // requires less bank angle.
+        // Formula based on http://aviation.stackexchange.com/a/8013
+        const turn_rate = clamp(0, 1 / (this.speed / 8.883031), 0.0523598776);
+        const turn_amount = turn_rate * window.gameController.game_delta();
+        const offset = angle_offset(this.target.heading, this.heading);
+
+        if (abs(offset) < turn_amount) {
+            this.heading = this.target.heading;
+        } else if ((offset < 0 && this.target.turn === null) || this.target.turn === 'left') {
+            this.heading -= turn_amount;
+        } else if ((offset > 0 && this.target.turn === null) || this.target.turn === 'right') {
+            this.heading += turn_amount;
         }
+    }
 
-        // ALTITUDE
-        let distance = null;
-        const expedite_factor = 1.5;
+    /**
+     * This updates the Altitude for the instance of the aircraft by checking the difference between current Altitude and requested Altitude
+     * @for AircraftInstanceModel
+     * @method updateAltitudePhysics
+     */
+    updateAltitudePhysics() {
         this.trend = 0;
 
         if (this.target.altitude < this.altitude - 0.02) {
-            distance = -this.model.rate.descent / 60 * window.gameController.game_delta();
+            this.decreaseAircraftAltitude();
+        } else if (this.target.altitude > this.altitude + 0.02) {
```

Comment:
is there a case where neither one of these ifs would get hit?  if there is, add another if block around line 1348 and return.  then you can probably remove the else if here.